File: garimpo-api/app/entitlements.py
"""Limites por plano (fonte única). Espelha `src/lib/entitlements.ts` do front."""

from dataclasses import dataclass


@dataclass(frozen=True)
class Entitlements:
    min_interval_minutes: float
    max_alerts: int
    max_destinations: int
    max_pages: int


_TABLE = {
    "FREE": Entitlements(1, 1, 1, 1),
    "PRO": Entitlements(0.25, 10, 3, 2),
    "ELITE": Entitlements(1 / 6, 30, 10, 3),
}


def entitlements_for(plan: str) -> Entitlements:
    return _TABLE.get(plan, _TABLE["FREE"])

File: garimpo-web/src/lib/entitlements.ts
import type { Plan } from "./api/types";

export interface Entitlements {
  minIntervalMinutes: number;
  maxAlerts: number;
  maxDestinations: number;
  maxPages: number;
  priority: "normal" | "alta" | "máxima";
  domains: number;
  historyDays: number | null;
}

const TABLE: Record<Plan, Entitlements> = {
  FREE: {
    minIntervalMinutes: 1,
    maxAlerts: 1,
    maxDestinations: 1,
    maxPages: 1,
    priority: "normal",
    domains: 1,
    historyDays: 7,
  },
  PRO: {
    minIntervalMinutes: 0.25,
    maxAlerts: 10,
    maxDestinations: 3,
    maxPages: 2,
    priority: "alta",
    domains: 3,
    historyDays: 90,
  },
  ELITE: {
    minIntervalMinutes: 1 / 6,
    maxAlerts: 30,
    maxDestinations: 10,
    maxPages: 3,
    priority: "máxima",
    domains: 10,
    historyDays: null,
  },
};

export function entitlementsFor(plan: Plan): Entitlements {
  return TABLE[plan];
}

export function nextPlan(plan: Plan): Plan | null {
  if (plan === "FREE") return "PRO";
  if (plan === "PRO") return "ELITE";
  return null;
}
